pkg/intermediate/aggregate.go

```diff
@@ -124,74 +140,272 @@ func (a *AggregationProcess) DeleteFlowKeyFromMap(flowKey FlowKey) {
+			}
 		}
 	}
-	if len(srcRecords) != 0 {
-		a.flowKeyRecordMap[flowKey] = srcRecords
-	} else {
-		a.flowKeyRecordMap[flowKey] = dstRecords
+
+	statsElementList := a.aggregateElements.statsElements
+	antreaSourceStatsElements := a.aggregateElements.aggregatedSourceStatsElements
+	antreaDestinationStatsElements := a.aggregateElements.aggregatedDestinationStatsElements
+	for i, element := range statsElementList {
+		isDelta := false
+		if strings.Contains(element, "Delta") {
+			isDelta = true
+		}
+		if ieWithValue, exist := incomingRecord.GetInfoElementWithValue(element); exist {
+			existingIeWithValue, _ := existingRecord.GetInfoElementWithValue(element)
+			// Update the corresponding element in existing record.
+			if !isDelta {
+				existingIeWithValue.Value = ieWithValue.Value
+			} else {
+				// We are simply adding the delta stats now. We expect delta stats to be
+				// reset after sending the record from flowKeyMap in aggregation process.
+				// Delta stats from source and destination nodes are added, so we will have
+				// two times the stats approximately.
+				// For delta stats, it is better to use source and destination specific
+				// stats.
+				existingIeWithValue.Value = existingIeWithValue.Value.(uint64) + ieWithValue.Value.(uint64)
+			}
+			// Update the corresponding source element in antreaStatsElement list.
+			if fillSrcStats {
+				existingIeWithValue, _ := existingRecord.GetInfoElementWithValue(antreaSourceStatsElements[i])
+				if !isDelta {
+					existingIeWithValue.Value = ieWithValue.Value
+				} else {
+					existingIeWithValue.Value = existingIeWithValue.Value.(uint64) + ieWithValue.Value.(uint64)
+				}
+			}
+			// Update the corresponding destination element in antreaStatsElement list.
+			if fillDstStats {
+				existingIeWithValue, _ := existingRecord.GetInfoElementWithValue(antreaDestinationStatsElements[i])
+				if !isDelta {
+					existingIeWithValue.Value = ieWithValue.Value
+				} else {
+					existingIeWithValue.Value = existingIeWithValue.Value.(uint64) + ieWithValue.Value.(uint64)
+				}
+			}
+		}
+	}
+}
+
+func (a *AggregationProcess) addFieldsForStatsAggregation(record entities.Record, fillSrcStats, fillDstStats bool) error {
+	if a.aggregateElements == nil {
+		return nil
+	}
+	statsElementList := a.aggregateElements.statsElements
+	antreaSourceStatsElements := a.aggregateElements.aggregatedSourceStatsElements
+	antreaDestinationStatsElements := a.aggregateElements.aggregatedDestinationStatsElements
+	antreaElements := append(antreaSourceStatsElements, antreaDestinationStatsElements...)
+
+	for _, element := range antreaElements {
+		// Get the new info element from Antrea registry.
+		// TODO: Take antrea registry enterpriseID as input to make this generic.
+		ie, err := registry.GetInfoElement(element, registry.AntreaEnterpriseID)
+		if err != nil {
+			return err
+		}
+		value := new(bytes.Buffer)
+		if err = util.Encode(value, binary.BigEndian, uint64(0)); err != nil {
```

Comment:
Just curious, does the 64-bit integer here cover InfoElement ID, field length, and enterprise number?

Reply:
It just initializes the value for new stat fields added for source and destination.